### PR TITLE
Move path action logic from view to model layer, add tests

### LINKED ISSUE #64: Move paths from asset view to asset model
Generally speaking, `@action`s on a view shouldn't really have any logic in them, they should simply farm that work out to the model layer.
Move the `AssetViewSet.paths` logic to a class method of the `Asset` model.

### PR DESCRIPTION
**Pull request by mvandenburgh**
Resolves #64.

### CODE REVIEW

File: publish/models/asset.py [hunk 1]
```diff
@@ -105,3 +105,22 @@ def from_girder(cls, version: Version, girder_file: GirderFile, client: GirderCl
             # The actual upload of blob occurs when the asset is saved
             asset.save()
         return asset
+
+    @classmethod
+    def get_path(cls, path_prefix: str, qs: str) -> set:
```
Comment (dchiquit) [inline publish/models/asset.py:110]: Isn't `qs` a list of str?

Comment (mvandenburgh) [reply publish/models/asset.py]: Yes, fixed in https://github.com/dandi/dandi-publish/pull/111/commits/23c77858bb09f292dfddf55ecc02b8e80331676b

**Comment by mgrauer** [comment]
Yo! can you add tests?

File: publish/models/asset.py [hunk 2]
```diff
@@ -105,3 +106,22 @@ def from_girder(cls, version: Version, girder_file: GirderFile, client: GirderCl
             # The actual upload of blob occurs when the asset is saved
             asset.save()
         return asset
+
+    @classmethod
+    def get_path(cls, path_prefix: str, qs: List[str]) -> set:
```
Comment (mgrauer) [inline publish/models/asset.py:111]: return capital s Set

File: dandi/publish/tests/test_asset.py [hunk 3]
```diff
@@ -73,3 +75,50 @@ def test_asset_rest_retrieve(api_client, asset):
         'updated': TIMESTAMP_RE,
         'metadata': asset.metadata,
     }
+
+
+@pytest.mark.django_db
+def test_asset_path(api_client, asset):
+    path = os.path.split(asset.path)
+    asset_directory = path[: len(path) - 1]
+    asset_filename = path[len(path) - 1]
```
Comment (dchiquit) [inline dandi/publish/tests/test_asset.py:84]: `os.path.split` always returns two strings, so you don't have to do this logic.
```suggestion
    asset_directory = path[0]
    asset_filename = path[1]
```

File: dandi/publish/tests/test_asset.py [hunk 4]
```diff
@@ -73,3 +75,50 @@ def test_asset_rest_retrieve(api_client, asset):
         'updated': TIMESTAMP_RE,
         'metadata': asset.metadata,
     }
+
+
+@pytest.mark.django_db
+def test_asset_path(api_client, asset):
```
Comment (dchiquit) [inline dandi/publish/tests/test_asset.py:81]: Part of the purpose of moving this logic to the model layer is that the logic no longer lives in the API layer, so you don't need to involve the API to test it. You should be able to directly invoke `asset.get_path('/foo',['/foo/bar', '/bar'])` to test the functionality of the `get_path` function, and then just have a couple of token API tests to verify that the `@paths` action was implemented correctly.

**Comment by mvandenburgh** [comment]
@dchiquit I can do it, shouldn't take too long

**Pull request merged**